traitsui/extras/tutor.py

```diff
@@ -873,729 +191,285 @@ def run_code(self):
-                    )
-            else:
-                self.section = Lecture(
-                    title=self.title,
-                    path=path,
-                    toc=self.toc,
-                    parent=self.parent,
-                    descriptions=self.descriptions
-                )
-        elif len(self.snippets) > 0:
-            self.section = Lab(
-                title=self.title,
-                path=path,
-                toc=self.toc,
-                parent=self.parent,
-                snippets=self.snippets,
-                auto_run=self.auto_run
-            )
-        else:
-            # No descriptions or code snippets were found. Create a lecture
-            # anyway:
-            section = Lecture(
-                title=self.title,
-                path=path,
-                toc=self.toc,
-                parent=self.parent
-            )
-
-            # If the lecture has subsections, then return the lecture and add
-            # a default item containing a description of the subsections of the
-            # lecture:
-            if len(section.subsections) > 0:
-                self._create_html_item(path=path, content=DefaultLecture % ('\n'.join(
-                    ['<li>%s</li>' % subsection.title
-                     for subsection in section.subsections])))
-                section.descriptions = self.descriptions
-                self.section = section
-
-    #-- Property Implementations ---------------------------------------------
-
-    def _get_css_path(self):
-        """ Returns the path to the CSS style sheet for the section.
-        """
-        return css_path_for(self.path, self.parent)
+        return self._parse_wmv_file(path, name)
 
-    #-- Factory Methods for Creating Section Items Based on File Type --------
-
-    def _add_py_item(self, path):
-        """ Creates the code snippets for a Python source file.
-        """
-        description, source = parse_source(path)
-        title = "Description"
-        self._add_description(description, title)
-        self.snippets.append(CodeItem(
-            title="Code",
-            path=path,
-            hidden=False,
-            content=source
-        ))
-
-    def _add_txt_item(self, path):
-        """ Creates a description item for a normal text file.
+    def _parse_htm_file(self, path, name):
+        """ Parse a htm file
         """
-        self.descriptions.append(TextItem(path=path))
+        html = _read_file(path)
+        return TutorialHTMLFile(description=html, name=name, parent=self)
 
-    def _add_htm_item(self, path):
-        """ Creates a description item for an HTML file.
+    def _parse_html_file(self, path, name):
+        """ Parse a html file
         """
-        # Check if there is a corresponding .rst (restructured text) file:
-        dir, base_name = os.path.split(path)
-        rst = os.path.join(dir, os.path.splitext(base_name)[0] + '.rst')
+        return self._parse_htm_file(path, name)
```

Comment:
This just does infinite recursion? (it looks like the original version did as well)

I think we can just remove this.